### PR TITLE
Add pre-commit for code style checks

### PR DESCRIPTION
**Pull request by Dunedan**
This adds the pre-commit framework for checking quality of committed
code and other files.

Additionally it fixes the parts of the code which didn't match the
configured rules. These changes are purely cosmetic and don't change
any logic in the code.

This commit also includes configuration for Github Actions to run the
checks for every pull request automatically.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,56 +1,64 @@
 # 0 A.D. / Pyrogenesis Multiplayer Lobby Setup
 
-This README explains how to setup a custom Pyrogenesis Multiplayer Lobby server that can be used with the Pyrogenesis game.
+This README explains how to setup a custom Pyrogenesis Multiplayer Lobby server that can be used
+with the Pyrogenesis game.
```
Comment (StanleySweet) [inline README.md:4]: I suppose engine would be more accurate than game?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# How to contribute
```
Comment (StanleySweet) [inline CONTRIBUTING.md:1]: This could be in its own  separate PR?

**Review by Oceandragon — APPROVED** [review]
This is good. 
README.md is kind of large with many changes. I skimmed it and didn't notice any changes or improvements to address.
Just noticed a couple spelling improvements in one of the tests.

File: tests/test_elo.py [hunk 3]
```diff
@@ -96,7 +96,7 @@ def test_sure_win(self, data):
         """Test behavior if winning player 1 has >600 points more.
 
         In this case the winning player shouldn't gain points, as it
-        was a "sure win" and the loosing player shouldn't loose
+        was a "sure win" and the losing player shouldn't loose
```
Comment (Oceandragon) [inline tests/test_elo.py:99]:
```suggestion
        was a "sure win" and the losing player shouldn't lose
```

File: tests/test_elo.py [hunk 4]
```diff
@@ -122,7 +122,7 @@ def test_sure_loss(self, rating_player1, difference_player2, played_games_player
         """Test behavior if winning player 2 has >600 points more.
 
         In this case the winning player shouldn't gain points, as it
-        was a "sure win" and the loosing player shouldn't loose
+        was a "sure win" and the losing player shouldn't loose
```
Comment (Oceandragon) [inline tests/test_elo.py:125]:
```suggestion
        was a "sure win" and the losing player shouldn't lose
```

**Review by Oceandragon — APPROVED** [review]

File: CONTRIBUTING.md (hunk 2 above)
Comment (Oceandragon) [reply CONTRIBUTING.md:1]: I think it seems correct to include the documentation about the pre-commit hook in the same commit that adds it.
Either way is OK. It might be that if logically 2 commits can be simply made into 1 commit, and it's just as clear, it's a bit better.

File: README.md (hunk 1 above)
Comment (Dunedan) [reply README.md:4]: I believe we should change the wording in general to make it clear that these bots are lobby bots for pyrogenesis (the engine), with 0ad (the game) as its main user. Changing this wording is probably a larger task, so I suggest we move that (and any possibly associated discussion) to a separate PR.

File: CONTRIBUTING.md (hunk 2 above)
Comment (Dunedan) [reply CONTRIBUTING.md:1]: I moved it to a separate commit. So it's one PR with two commits now. :slightly_smiling_face:

File: tests/test_elo.py (hunk 3 above)
Comment (Dunedan) [reply tests/test_elo.py]: Great finding. :+1:
I fixed that now.

File: tests/test_elo.py (hunk 4 above)
Comment (Dunedan) [reply tests/test_elo.py]: Great finding. :+1:
I fixed that now.
